### PR TITLE
Teleport session improvements

### PR DESCRIPTION
**Pull request by kontsevoy**
Two changes:
- e349dcc - Make sure all active sessions are properly closed when a `server.Close()` is called.
- 8009984 - Replay last chunk to a new party who joins an existing session.

### CODE REVIEW

File: lib/srv/sess.go [hunk 1]
```diff
@@ -419,11 +426,39 @@ func (s *session) addParty(p *party) {
...
 
-const partyTTL = 10 * time.Second
-
 func (s *session) join(sconn *ssh.ServerConn, ch ssh.Channel, req *ssh.Request, ctx *ctx) (*party, error) {
 	p := newParty(s, sconn, ch, ctx)
 	s.addParty(p)
+
+	// when a new user joins the session, it's visually helpful to replay the last piece of output so they
+	// will see the same prompt the host is seeing
+	replayLastChunk := func() {
```
Comment (klizhentas) [inline lib/srv/sess.go]: can you move this function to be a method? seems a bit generic to be a closure

**Comment by klizhentas** [comment]
lgtm with comment about a closure/method

File: lib/srv/sess.go (hunk 1 above)
Comment (kontsevoy) [inline lib/srv/sess.go]: mm... actually I think this must be deleted alltogether. to be discussed.

**Comment by kontsevoy** [comment]
I changed the implementation. See the comments for f81bfdb

**Comment by klizhentas** [comment]
lgtm

**Pull request merged**